pkg/config/tokens.go

```diff
@@ -0,0 +1,84 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/coinbase/rosetta-geth-sdk/configuration"
+)
+
+// ReadTokenConfig attempts to read file contents from a filename.
+func ReadTokenConfig(filename string) (string, error) {
+	contents := ""
+	if file, err := os.ReadFile(filename); err == nil {
+		// if the envvar points to a file, read it; otherwise the envvar contents is expected to be JSON
+		contents = string(file)
+	}
+	if contents == "" {
+		contents = "[]"
+	}
+	return contents, nil
+}
+
+// UnmarshalTokenConfig attempts to construct a list of [configuration.Token] from a JSON file.
+// Accepts variadic network IDs used to filter tokens.
+func UnmarshalTokenConfig(contents []byte, networks ...uint64) ([]configuration.Token, error) {
+	// Try to parse the file as a list of tokens.
+	var payload []configuration.Token
+	if err := json.Unmarshal(contents, &payload); err == nil {
+		return FilterNetworks(payload, networks...), nil
+	}
+
+	// If this fails, try the backwards-compatible token json format
+	var outer map[string]interface{}
+	if err := json.Unmarshal(contents, &outer); err == nil {
+		for k, v := range outer {
+			for t, b := range v.(map[string]interface{}) {
+				if b.(bool) {
+					switch strings.ToLower(k) {
+					case "mainnet":
+						payload = append(payload, configuration.Token{
+							ChainID: 1,
+							Address: t,
+						})
+					case "testnet":
+						payload = append(payload, configuration.Token{
+							ChainID: 420,
```

Comment:
nit: rather than hard code the chain ID, you can use the ChainID in the `CHAIN_CONFIG` env.